static/src/javascripts/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -0,0 +1,263 @@
+    // key = 'surveyToShow', and value = the survey id. Tailor will then override other logic for display, and
+    // look for a survey with this ID to return. This is useful as we can easily see how a particular survey
+    // would be rendered, without actually putting it live. If this parameter is empty or not specified, tailor
+    // behaves as usual.
+    const surveyToShow = local.get('surveyToShow');
+
+    if (surveyToShow) {
+        queryParams.surveyToShow = surveyToShow;
+    }
+
+    // get the list of surveys that can't be shown as they have been shown recently
+    const surveysNotToShow = getSurveyIdsNotToShow();
+
+    if (surveysNotToShow) {
+        queryParams.surveysNotToShow = surveysNotToShow;
+    }
+
+    return tailor.getSuggestedSurvey(queryParams).then(suggestion => {
+        if (suggestion) {
+            storeSurveyShowedInCookie(suggestion.data);
+
+            const json = getJsonFromSurvey(suggestion.data.survey);
+            const survey = bonzo.create(template(tailorSurveyHtml, json));
+
+            return inArticleWriter(survey, suggestion.data.survey.surveyId);
+        }
+        Promise.resolve();
+    });
+};
+
+const disableRadioButtons = buttonClassName => {
+    const radioButtons = document.getElementsByClassName(buttonClassName);
+    bonzo(radioButtons).each(button => {
+        // eslint-disable-next-line no-param-reassign
+        button.disabled = true;
+    });
+};
+
+const surveyFadeOut = () => {
+    const surveyContent = document.getElementsByClassName(
+        'impressions-survey__content'
+    );
+
+    surveyContent[0].classList.add('js-impressions-survey__fadeout');
+};
+
+const thankyouFadeIn = () => {
+    const surveyThanks = document.getElementsByClassName(
+        'impressions-survey__thanks'
+    );
+
+    surveyThanks[0].classList.add('js-impressions-survey__fadein');
+};
+
+const recordOphanAbEvent = (answer, surveyId) => {
+    const componentId = `data_tailor_survey_${surveyId}`;
+
+    ophan.record({
+        component: componentId,
+        value: answer,
+    });
+};
+
+const handleSurveyResponse = surveyId => {
+    const surveyQuestions = [
+        ...document.getElementsByClassName('fi-survey__button'),
+    ];
+
+    surveyQuestions.forEach(question => {
+        bean.on(question, 'click', event => {
+            if (event.target.attributes.getNamedItem('data-link-name')) {
+                const answer = event.target.attributes.getNamedItem(
+                    'data-link-name'
+                ).value;
+
+                recordOphanAbEvent(answer, surveyId);
+
+                mediator.emit('tailor:survey:clicked');
+                fastdom.write(() => {
+                    disableRadioButtons('fi-survey__button');
+                    surveyFadeOut();
+                    thankyouFadeIn();
+                });
+            }
+        });
+    });
+};
+
+class TailorSurvey {
```

Comment:
Yeah, I don't necessarily agree with the current approach of creating AB test objects with the `new` keyword. You can totally create your AB test in `javascripts`, in fact I highly recommend that you do.

I don't want to change too much in this PR. Perhaps for future tests, we can recommend using the immutable object approach. Maybe I'll start by updating the docs.